### ISSUE TITLE
Security.do_not_log_authkeys has no effect

### ISSUE THREAD

**Issue by cudeso** [MISP/MISP#7093]
### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       | Ubuntu
| MISP version / git hash   | 2.4.139

### Expected behavior

Log API key when **do_not_log_authkeys** is set to false

### Actual behavior

Regardless if set to false or true, the API key logged is always

`Successful authentication using API key (aBCD********************************abcd) `

**Comment by JakubOnderka** [comment]
Hello, this is intended. This option prevents agains logging the whole key, but logging just first and last 4 characters is considered safe.

**Comment by cudeso** [comment]
I understand. But isn't it better to remove the option completely? Now setting it either to true or false returns the same result.

**Comment by cudeso** [comment]
To clarify: either if the option is set to true or false, the logged output is **always** "(aBCD********************************abcd)"

**Comment by lfortemps** [comment]
I think this option is useful for people that haven't turned on the advanced auth key setting yet.

**Comment by cudeso** [comment]
Yes. But as far that I can check, the only option having effect is log_user_ips_authkeys (either yes/no log of API requests) and do_not_log_authkeys has no effect (always logged with aBCD*** regardless of setting)

**Comment by Wachizungu** [comment]
It indeed doesn't seem to make a difference when authentication is logged. The do_not_log_authkeys does have an effect still. When you reset a key using the old auth key system, the full key is logged if do_not_log_authkeys is false. You only see asterisks in the log for those entries if do_not_log_authkeys is true.
